### PR TITLE
Proper initialization of DLInfo

### PR DESCRIPTION
**Pull request by khagankhan**
Fixed https://github.com/WasmEdge/WasmEdge/issues/3014

### CODE REVIEW

**Review by hydai — CHANGES_REQUESTED** [review]
According to the man page:
```
RETURN VALUE         [top](https://man7.org/linux/man-pages/man3/dladdr.3.html#top_of_page)
       On success, these functions return a nonzero value.  If the
       address specified in addr could be matched to a shared object,
       but not to a symbol in the shared object, then the
       info->dli_sname and info->dli_saddr fields are set to NULL.

       If the address specified in addr could not be matched to a shared
       object, then these functions return 0.  In this case, an error
       message is not available via [dlerror(3)](https://man7.org/linux/man-pages/man3/dlerror.3.html).
```

The error message should be more accurate.

**Comment by dannypsnl** [comment]
Please read https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

**Comment by khagankhan** [comment]
Oh, I see! I will sign it off now when adding detailed error message.

**Comment by khagankhan** [comment]
It is almost done. If anything needed please let me know - "we've come too far to give up".

Thank you for your guidance and the chance to work on this issue! o7

File: lib/plugin/plugin.cpp [hunk 1]
```diff
@@ -272,6 +277,11 @@ std::vector<std::filesystem::path> Plugin::getDefaultPluginPaths() noexcept {
       Result.push_back(LibPath / std::filesystem::u8path(".."sv) /
                        std::filesystem::u8path("plugin"sv));
     }
+  } else {
+    spdlog::error(ErrCode::Value::NonNullRequired);
+    spdlog::error("Address could not be matched to any shared object. "
+                  "Detailed error information is not available.");
```
Comment (hydai) [inline lib/plugin/plugin.cpp:283]: Use `"..."sv` instead of a normal string, like line 278 `"plugin"sv`.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]

**Pull request merged**